shacl/DESIGN.md

```diff
@@ -0,0 +1,92 @@
+# SHACL for Brick
+
+The SHACL Shapes Constraint Language validates RDF graphs against conditions (called "shapes"), which are
+expressed themselves as an RDF graph. This document discusses the application of SHACL to Brick. In
+this document we are specifically looking at writing SHACL shapes for validating Brick *models*
+(instances of buildings). Using SHACL to validate the Brick ontology deserves its own investigation.
+
+SHACL shapes for Brick should validate the following:
+
+- validating endpoints of Brick relationships (properties):
+  - check type of subject/object: are the `brick:expectedDomain` and `brick:expectedRange` properties respected?
+  - cardinality checks: with varying severity
+    - floors without rooms
+    - equipment missing certain points (e.g. thermostat should have a temperature sensor, at least one setpoint)
+    - all named individuals should have `rdf:type`
+- compatibility checking:
+  - all objects of `rdf:type` should be Brick classes (subclassof, etc)
+  - check for deprecated classes, relationships
+- idioms, semantic checks:
+  - these may be optional; users can "opt-in" to check for certain idioms
+  - equipment along a chain of `brick:feeds` should all operate on the same substance
+  - proper relationships exist between equipment: e.g. thermostat should have `controls` relationship to RTU
+
+## Getting Started
+
+Install [`pyshacl`](https://github.com/RDFLib/pySHACL) from pip.  Run
+ `./validate.sh <building graph file>` to
+* generate ontology file Brick.ttl
+* generate shapes grapph shacl_test.ttl
+* execute a wrapper of the `pyshacl` tool to validate the building graph and report violations
+
+As an example, try `./validate.sh sample_graph.ttl`.
+
+## Brick SHACL shapes
+
+Those are the rules focusing on certain type of triples in the building graph and enforcing restrictions.  The
+examples below declare shapes in the `bsh` ("brickshape") namespace for clarity.
```

Comment:
Can you include the full URI of the `bsh` namespace as well, for clarity?

Reply:
done